core/transactions-factories/token_management_transactions_factory.md

```diff
@@ -97,90 +97,164 @@ class TokenManagementTransactionsFactory:
         addRoleNFTBurn: boolean;
         addRoleNFTAddQuantity: boolean;
         addRoleESDTTransferRole: boolean;
+        addRoleESDTModifyCreator: boolean;
     }): Transaction;
 
     create_transaction_for_setting_special_role_on_non_fungible_token({
         sender: IAddress;
         user: IAddress;
-        tokenIdentifier: str;
+        tokenIdentifier: string;
         addRoleNftCreate: bool;
         addRoleNftBurn: bool;
         addRoleNftUpdate_attributes: bool;
         addRoleNftAddUri: bool;
         addRoleESDTTransferRole: bool;
+        addRoleESDTModifyCreator: boolean;
+        addRoleNFTRecreate: boolean;
     }): Transaction;
```

Comment:
i think we need here some more roles, like `ESDTRoleSetNewURI`, `ESDTRoleModifyRoyalties`

Reply:
do we intend to have a function for setting custom roles for token?

Reply:
Completely custom / arbitrary strings - I'd vote no, but only for now - maybe that will be nice in the future.

Reply:
added